msteams-platform/tabs/how-to/personal-tab-navigation.md

```diff
@@ -0,0 +1,63 @@
+
+## Navigate between tabs
+
+The [pages](/javascript/api/@microsoft/teams-js/pages?view=msteams-client-js-latest&preserve-view=true) capability of the TeamsJS library provides support for navigation between tabs within a personal app. Specifically, the [`pages.currentApp`](/javascript/api/@microsoft/teams-js/pages.currentapp?view=msteams-client-js-latest&preserve-view=true) namespace offers a function `navigateTo(NavigateWithinAppParams)` to allow navigation to a specific tab within the current app and a function `navigateToDefaultPage()` to navigate to the first tab defined in the app's manifest.
+
+The following code illustrates how to navigate to a specific page:
+
+```js
+if (pages.currentApp.isSupported()) {
+    const navPromise = pages.currentApp.navigateTo({pageId: <pageId>, subPageId:<subPageId>});
+    navPromise.
+        then((result) => {/*Successful navigation*/}).
+        catch((error) => {/*Failed navigation*/});
+}
+else {/*Handle situation where capability isn't supported*/}
+```
+
+The following code illustrates how to navigate to the app's default tab:
+
+```js
+if (pages.currentApp.isSupported()) {
+    const navPromise = pages.currentApp.navigateToDefaultPage();
+    navPromise.
+        then((result) => {/*Successful navigation*/}).
+        catch((error) => {/*Failed navigation*/});
+}
+else {/*Handle situation where capability isn't supported*/}
+```
+
+You can also use the `pages.navigateToApp()` function, `pages.tabs.navigateToTab()` function, or deep links for tab navigation, as shown in [Create deep links](~/concepts/build-and-test/deep-links.md#navigate-within-your-app).
+
+## Configure back button navigation
+
+When a personal app has multiple tabs, a user can use the Microsoft 365 host app's back button to go backwards through the navigational history. However that history doesn't include actions a user performs within a tab. If you need to enhance the back button experience, you can maintain your own internal navigation stack and configure a custom handler for back button selections. This is accomplished through the `registerBackButtonHandler()` function in the [`pages.backStack`](/javascript/api/@microsoft/teams-js/pages.backstack?view=msteams-client-js-latest&preserve-view=true) namespace.
+
+After you register the handler, it helps you to address the navigational request before the system takes action. If the handler is able to manage the request, it is set to `true` so that the system knows no further action is necessary. If the internal stack is empty, it is set to `false` so that the system can call the `navigateBack()` function instead and take the appropriate action.
```

Comment:
```suggestion
After you register the handler, it helps you to address the navigational request before the system takes action. If the handler is able to manage the request, it's set to `true` so that the system knows no further action is necessary. If the internal stack is empty, it's set to `false` so that the system can call the `navigateBack()` function instead and take the appropriate action.
```